### PR TITLE
🚧Create new solution service

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">✅Code Review Checklist✅</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/feature/<issue-num>-<description>
      Example: preview/feature/123-my-preview-feature
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_preview release_** branch, with the correct naming syntax, in the following example:
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
    Example: preview/v1.2.3-preview.4
    ```
  </details>

- [x] The **_base(target)_** branch for this pull request is a **_preview release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

💡For more information on branching, refer to the project's [branching documentation](../../Documentation/Branching.md).

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title matches the title of the associated issue.

- [x] A `preview` label has been added to the pull request.

- [ ] The labels attached to this PR match the labels attached to the associated issue.

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out and let's discuss it.

- [x] I have manually tested my code changes to the best of my ability.

- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - Status checks run for this pull request when merging into the **_preview/v\*.\*.\*-preview.\*_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

File: CICD/CICD.cs [hunk 1]
```diff
@@ -31,10 +30,34 @@ public partial class CICD : NukeBuild
     /// The main entry point of the build system.
     /// </summary>
     /// <returns>An <c>integer</c> value representing an error code or 0 for no errors.</returns>
-    public static int Main() =>
-        Execute<CICD>(x => x.BuildAllProjects, x => x.RunAllUnitTests);
+    public static int Main() => Execute<CICD>(x => x.BuildAllProjects, x => x.RunAllUnitTests);
 
 #pragma warning disable SA1201 - A property should not follow a method
```
Comment (kselena) [inline CICD/CICD.cs:35]: Check warnings about single line comments and fix.

File: CICD/ISolutionWrapper.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    /// Gets a project that matches the given project name or fully qualified path.
+    /// </summary>
+    /// <param name="nameOrFullPath">The name of the project or the fully qualified project file path to the <c>csproj</c> file.</param>
+    /// <returns>A solution project.</returns>
+    Project? GetProject(string nameOrFullPath);
+
+    /// <summary>
+    /// Returns a list of projects that match the given <paramref name="wildcardPattern"/>.
+    /// </summary>
+    /// <param name="wildcardPattern">The pattern to use to find solution projects.</param>
```
Comment (kselena) [inline CICD/ISolutionWrapper.cs:36]: Grammar/verbose
```suggestion
    /// <param name="wildcardPattern">The pattern to find solution projects.</param>
```

**Review by KinsonDigitalAdmin — APPROVED** [review]

File: CICD/Services/SolutionService.cs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            ? version.TrimStart("v")
+            : version;
+
+        var fileName = $"Release-Notes-v{version}.md";
+
+        return $"{notesDirPath}/{fileName}";
+    }
+
+    /// <inheritdoc/>
+    // TODO: Add exceptions
```
Comment (kselena) [inline CICD/Services/SolutionService.cs:73]: Do you need this TODO?

File: Testing/CICDTests/Services/SolutionServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+        var actual = sut.AllProjects;
+
+        // Assert
+        actual.Should().BeEquivalentTo(expected);
+    }
+    #endregion
+
+    #region Method Tests
+    [Fact]
+    public void GetProject_WhenInvoked_ReturnsCorrectResult()
```
Comment (kselena) [inline Testing/CICDTests/Services/SolutionServiceTests.cs:98]: I believe this method should be GetProjects.  That is what it's called above this block and below this block.
●
```suggestion
    public void GetProject_WhenInvoked_ReturnsCorrectResult()
```

**Review by kselena — CHANGES_REQUESTED** [review]

File: Testing/CICDTests/Services/SolutionServiceTests.cs (hunk 4 above)
Comment (CalvinWilkinson) [reply Testing/CICDTests/Services/SolutionServiceTests.cs:98]: Nope,  this is correct.  I have 2 methods that get tested.  `GetProject()` and `GetProjects()`.
Look at lines 109 and 132.  These are the actual methods(functions) being executed for testing purposes.

**Review by kselena — APPROVED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]
